src/lib/optimizer/abstract_syntax_tree/aggregate_node.hpp

```diff
@@ -64,8 +64,9 @@ class AggregateNode : public AbstractASTNode {
   std::vector<std::shared_ptr<Expression>> _aggregate_expressions;
   std::vector<ColumnID> _groupby_column_ids;
 
-  std::vector<ColumnID> _output_column_id_to_input_column_id;
-  std::vector<std::string> _output_column_names;
+  mutable std::vector<std::string> _output_column_names;
```

Comment:
Not a fan of naming it cache (its not a compute-if-key-is-missing-otherwise-return-value-thing) not a fan of wrapping it in optional, but will if you insist. In any case, doc'ed it.

Reply:
Maybe not "cache", but I think the name should make it clear that this is potentially incorrect.

For what reason do you prefer it not to be an optional?

Reply:
I've seen the `_lazy_*` prefix being used, I've also seen `template<typename T> using LazyInitialized = std::optional<T>` I think... anyway! Don' really have anything against optional, just feels like a wasted byte.

Proposed solution: std::optional. 

We've got other lazy initialized variables (AST node statistics come to mind, e.g.), so if you want a name change, maybe deferring that to an issue that adjusts all places would be the better solution. No name changes needed from my side, though I would be fine with it.

Reply:
If it's just about the byte, I would go with the optional, because it make clear that `(bool)_output_column_names` is not the same as `_output_column_names->empty()`.

Agreed on moving that into an issue. Otherwise people might think that they could modify the output column names by writing into that vector.